### DISCUSSION TITLE
Clear single item from undo manager

### DISCUSSION THREAD

**Discussion by mihir-khandekar** [GrapesJS/grapesjs#3897] [Ideas]
Hello @artf ,

use case: We have a custom set of buttons to allow someone to undo/redo the last change. Changes like selecting a specific type of element from being selected are not expected to be tracked. Although, these components are marked as `data-gjs-selectable="false"` the editor still fires the component selected event and the undo manager adds this selection to its stack.

I would like to make a request where in we can remove a single tracked change from the undo manager using a method like `UndoManager.clear(changeId)`.

Please let me know if this can be done or if there are any work arounds.

**Comment by artf** [comment]
> the editor still fires the component selected event and the undo manager adds this selection to its stack.

I think you should investigate that first. When you click on the non-selectable component it tries to find the closest selectable one, so I guess you're just selecting the parent component.

**Reply by mihir-khandekar** [reply]
Hi, thanks for the update. Is there a way to prevent/disable this behavior?

>  When you click on the non-selectable component it tries to find the closest selectable one.

**Reply by artf** [reply]
No, otherwise you won't be able to select components behind the unselectable one.

**Reply by mihir-khandekar** [reply]
Okay, sounds reasonable.

Is there any way to modify the undo manager stack? I would like to clear the items in the stack conditionally. Any way to do this?

**Reply by artf** [reply]
Not really but I'm planning to improve the UndoManager module API soon. Subscribe to this https://github.com/artf/grapesjs/issues/3639

**Reply by mihir-khandekar** [reply]
Great. Please let me know if I can help in any way. Happy to contribute :)
